flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobAlreadyDoneException.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.jobmaster;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.JobException;
+
+/** Exception indicating that a Flink job failed in finishing phase. */
+public class JobAlreadyDoneException extends JobException {
+    private static final long serialVersionUID = 5906282500322681417L;
+
+    public JobAlreadyDoneException(JobID jobId) {
+        super("The job (" + jobId + ") has already terminated.");
```

Comment:
I think we should add that we cannot tell the final state and also not serve the `JobResult`. Consequently, the user would have to check the result of the job manually (e.g. whether output files are there).